### ISSUE TITLE
Timeframe for move to v3.x and availability of new keys

### ISSUE THREAD

**Issue by imagico** [gravitystorm/openstreetmap-carto#1243]
The roadmap indicates current development work in v2.x is focussed on fixing problems and internal redesign with only smaller changes to the output.  Many recent changes have however made quite large modifications to the appearance of the style, in particular for example #747, #941 and #1153.  So with regards to the development priorities outlined in the roadmap we are definitely already in the v3.x domain to some extent.

With this background there are many situations where tags currently not available in the rendering database are seriously needed for rendering, in other words there are many widely used tags that could be well used in rendering that are not currently available and this severely limits the style's ability to properly show the work of mappers.

The questions here are

* what are the current plans as to when and how the formal move to v3.x is going to happen?
* what are the factors holding this back, like are there technical requirements that are not met like in terms of the hardware?
* in what form is this going to happen, will there be a simple update to the fixed list of tags or will there be a dynamic approach without a limitation to a fixed list?

I realize there might not yet be answers to these questions but if not it might be time working on them.

**Comment by nebulon42** [comment]
+1 for bringing this up. I assumed this was also somehow tied to the availability of Mapnik 3.x, which is now close ahead (at least there was a first RC).

**Comment by math1985** [comment]
I think this is a question for @gravitystorm and @pnorman.

**Comment by HolgerJeromin** [comment]
This is probably also a question for the helping hands behind the rendering database. 
If i understood correct we need a full reimport of the OSM database with the hstore to have the full (?) osm madness for rendering available.
Perhaps we have an reimport in the mid term future (for hardware reasons?) where we could activate the hstore and after that we can start using this.

**Comment by pnorman** [comment]
> I assumed this was also somehow tied to the availability of Mapnik 3.x, which is now close ahead (at least there was a first RC).

@springmeyer says it will be at least March for Mapnik 3 to be released. Mapnik 3 is a hard requirement for some fixes, and I had hoped it would be out for 3.0 of the style so we could start using its features for other enhancements. If the release drags on, then maybe 4.0 of the style will start to use Mapnik 3 features.

I would say the following are blocked by not having a Mapnik 3 release

* Road shield improvements
* SVGs on anything other than POIs
* Support for non-European languages
* Rotating symbols

-----

Right now I would place fixing some of the remaining big style issues (e.g. roads colours) as more important than hstore.

Doing hstore right is complicated. At the same time as introducing hstore, the number of fixed columns should be reduced, and this should lead to a noticeable performance gain, but this requires benchmarking work and query rewrites. Operationally it requires some planning too, as it requires a DB reload, but that's the responsibility of the OWG and sysadmins.

**Comment by HolgerJeromin** [comment]
Perhaps we should split the milestone "3.x - Needs upgrade to mapnik or openstreetmap-carto.style" to "Needs upgrade to mapnik 3.x" and "Needs upgrade of openstreetmap-carto.style".

**Comment by pnorman** [comment]
I split the milestones - some issues might need reassigning.

**Comment by math1985** [comment]
Superseded by #1504 which is more concrete.
